.github/composite_actions/log_metric/action.yaml

```diff
@@ -0,0 +1,52 @@
+name: 'Log Metric'
+description: 'Log data point to a metric with the provided value. If the metric is not there, it will create one.'
+inputs:
+  aws-region:
+    required: true
+    description: The AWS region
+  role-to-assume:
+    required: true
+    description: The role to assume in the STS session
+  metric-name:
+    description: 'target metric name'
+    required: true
+  value:
+    description: 'the value to log to the metric'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@67fbcbb121271f7775d2e7715933280b06314838 # 1.7.0
+      with:
+        role-to-assume: ${{ inputs.role-to-assume }}
+        aws-region: ${{ inputs.aws-region }}
+        role-duration-seconds: 900
+
+    - name: Create AWS profile
+      run: ./build-support/create_aws_test_profile.sh
+      shell: bash
+
+    - name: Put metric data
+      shell: bash
+      # Run bash script to put metric data.
+      # First trim the input by removing trailing or leading spaces, then removing any new line characters.
+      # Then, we do regex test to match against expected input shape. Finally, put metric data using aws cli.
+      run: |
+        metric_name_trimmed="$(echo -e "${metric_name}" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//' | tr -d '\n')"
```

Comment:
Maybe use Dart for this? This is very dense and IIUC is just doing the equivalent of `string.trim()`